daemon/src/wallpaper_info.rs

```diff
@@ -12,6 +12,8 @@ pub struct WallpaperInfo {
     pub duration: Option<Duration>,
     #[serde(rename = "apply-shadow")]
     pub apply_shadow: Option<bool>,
+    #[serde(default = "set_sorting_default", deserialize_with = "order_deserializer")]
+    pub sorting: Option<String>
```

Comment:
If this were an `enum Sorting` which implemented `Deserialize`, `serde` could deserialize it directly:

```rust
    #[serde(default = "set_sorting_default")]
    pub sorting: Option<Sorting>,
```

If `Sorting` additionally implemented [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html), you can drop the `Option<T>` and tell `serde` to call `Sorting::default()` if it's not specified:

```rust
    #[serde(default)]
    pub sorting: Sorting,
```

You can `#[derive]` both `Deserialize` and `Default` along with all the other traits which might be useful:

```rust
#[derive(Debug, Copy, Clone, Default, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename = "kebab-case")]
pub enum Sorting {
    #[default]
    Random,
    Natural,
    Reverse,
}
```

Reply:
Fantastic! I was curious how to do that. String comparisons are my least favorite. I've absolutely done this, almost verbatim. :+1: